openff/qcsubmit/validators.py

```diff
@@ -288,14 +286,22 @@ def check_environments(environment: str) -> str:
     """
 
     # try and make a new chemical environment checking for parse errors
-    _ = ChemicalEnvironment(smirks=environment)
-
-    # check for numeric tags in the environment
-    if re.search(":[0-9]]+", environment) is not None:
-        return environment
-
-    else:
-        raise SMIRKSParsingError(
-            "The smarts pattern passed had no tagged atoms please tag the atoms in the "
-            "substructure you wish to include/exclude."
-        )
+    try:
+        _ = Molecule().chemical_environment_matches(environment)
+        # check for numeric tags in the environment
+        if re.search(":[0-9]]+", environment) is not None:
+            return environment
+    except ValueError as e:
+        # only catch an error like 'No registered toolkits can provide the
+        # capability "find_smarts_matches" for args...' it would be nice for
+        # chemical_environment_matches to raise a more specific exception, but
+        # it just raises a ValueError
+        if 'capability "find_smarts_matches"' not in str(e):
+            raise e
```

Comment:
Currently, this means that if we don't have any valid toolkits installed, the function falls down to the `SMIRKSParsingError` below about not having any tagged atoms. Would it make sense to still raise this toolkit capability error to communicate that scenario?

Reply:
Hmm, I see what you mean. Is there a better way to differentiate between them than checking this string? Ideally I would only except SMIRKSParsingError, but if the underlying code doesn't differentiate I wasn't sure how to do better than this. We could certainly log a warning at the very least.